packages/@aws-cdk/aws-pipes-alpha/lib/logs.ts

```diff
@@ -93,3 +153,81 @@ export interface ILogDestination {
    */
   grantPush(grantee: IRole): void;
 }
+
+/**
+ * CloudWatch Logs log group for delivery of pipe logs.
+ */
+export class CloudwatchLogsLogDestination implements ILogDestination {
+  private logGroup: ILogGroup;
+
+  constructor(logGroup: ILogGroup) {
+    this.logGroup = logGroup;
+  }
+
+  bind(_pipe: IPipe): LogDestinationConfig {
+    return {
+      parameters: {
+        cloudwatchLogsLogDestination: {
+          logGroupArn: this.logGroup.logGroupArn,
+        },
+      },
+    };
+  }
+
+  grantPush(pipeRole: IRole): void {
+    this.logGroup.grantWrite(pipeRole);
+  }
+}
+
+/**
+ * Firehose stream for delivery of pipe logs.
+ */
+export class FirehoseLogDestination implements ILogDestination {
+  private deliveryStream: IDeliveryStream;
+
+  constructor(deliveryStream: IDeliveryStream) {
+    this.deliveryStream = deliveryStream;
+  }
+
+  bind(_pipe: IPipe): LogDestinationConfig {
+    return {
+      parameters: {
+        firehoseLogDestination: {
+          deliveryStreamArn: this.deliveryStream.deliveryStreamArn,
+        },
+      },
+    };
+  }
+
+  grantPush(pipeRole: IRole): void {
+    this.deliveryStream.grantPutRecords(pipeRole);
+  }
+}
+
+/**
+ * S3 bucket for delivery of pipe logs.
+ */
+export class S3LogDestination implements ILogDestination {
+  private parameters: S3LogDestinationProps;
+
+  constructor(parameters: S3LogDestinationProps) {
+    this.parameters = parameters;
+  }
+
+  bind(_pipe: IPipe): LogDestinationConfig {
+    return {
+      parameters: {
+        s3LogDestination: {
+          bucketName: this.parameters.bucket.bucketName,
+          bucketOwner: this.parameters.bucketOwner || this.parameters.bucket.env.account,
```

Comment:
I had to add this default otherwise I got the following:

```
Failed resources:
aws-cdk-pipes | 10:24:59 AM | UPDATE_FAILED        | AWS::Pipes::Pipe                     | Pipe (PipeXXX) Resource handler returned message: "[object has missing required properties (["BucketOwner"])] (Service: Pipes, Status Code: 400, Request ID: xxx)" (RequestToken: xxx, HandlerErrorCode: InvalidRequest)
```

Add this to the doc: `@default - account ID derived from `bucket``

Reply:
Shouldn't be an issue given how the `env` prop is built (see below), but just for safety could you add a bucket import in a unit test?

https://github.com/aws/aws-cdk/blob/5cb47dabec40b22140f8538275dbdc2780028d04/packages/aws-cdk-lib/core/lib/resource.ts#L171-L174

Reply:
Great shout. I added a unit test here: https://github.com/aws/aws-cdk/pull/31672/files#diff-7ee5baaeb407c0ce0a9d967e0d0ef9bd3523ede78b48d34fea55b538b030e739R564.